deps/v8/src/libplatform/default-platform.cc

```diff
@@ -34,6 +34,18 @@ const int DefaultPlatform::kMaxThreadPoolSize = 8;
 DefaultPlatform::DefaultPlatform()
     : initialized_(false), thread_pool_size_(0) {}
 
+void DefaultPlatform::ForkingCleanup() {
+  base::LockGuard<base::Mutex> guard(&lock_);
+  queue_.PurgeWorkers();
+  if (initialized_) {
+    for (auto i = thread_pool_.begin(); i != thread_pool_.end(); ++i) {
+      delete *i;
+    }
+  }
+  thread_pool_.clear();
+  queue_.StopPurgeWorkers();
```

Comment:
@mmorearty just to document our discussion. In the Thread's destructor, we join on the corresponding pthread. This means we should never reach StopPurgeWorkers until all Pthreads have exited.